packages/transaction-controller/src/TransactionController.ts

```diff
@@ -377,10 +396,14 @@ export class TransactionController extends BaseController<
       this.failTransaction(transactionMeta, error);
       return Promise.reject(error);
     }
-
-    transactions.push(transactionMeta);
-    this.update({ transactions: this.trimTransactionsForState(transactions) });
-    this.hub.emit(`unapprovedTransaction`, transactionMeta);
+    // Checks if a transaction already exists with a given actionId
+    if (!existingTransactionMeta) {
+      transactions.push(transactionMeta);
+      this.update({
+        transactions: this.trimTransactionsForState(transactions),
+      });
+      this.hub.emit(`unapprovedTransaction`, transactionMeta);
+    }
 
     return {
       result: this.processApproval(transactionMeta),
```

Comment:
A bit confused here, shouldn't we somehow skip creating another approval if it's an `existingTransaction`?